slick/src/test/java/akka/stream/alpakka/slick/javadsl/DocSnippetFlow.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package example;
+
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import akka.Done;
+import akka.actor.ActorSystem;
+import akka.stream.ActorMaterializer;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+
+import akka.stream.alpakka.slick.javadsl.*;
+
+//#flow-example
+public class DocSnippetFlow {
+  public static void main(String[] args) throws Exception {
+    final ActorSystem system = ActorSystem.create();
+    final Materializer materializer = ActorMaterializer.create(system);
+
+    final SlickSession session = SlickSessions.forConfig("akka.stream.alpakka.slick.h2");
+
+    final List<User> users = IntStream.range(0, 42).boxed().map((i) -> new User(i, "Name"+i)).collect(Collectors.toList());
+
+    final CompletionStage<Done> done =
+      Source
+        .from(users)
+        .via(
+          Slick.<User>flow(
+            session,
+            // add an optional second argument to specify the parallism factor (int)
+            (user) -> "INSERT INTO ALPAKKA_SLICK_JAVADSL_TEST_USERS VALUES (" + user.id + ", '" + user.name + "')"
+          )
+        )
+        .log("nr-of-updated-rows")
+        .runWith(Sink.ignore(), materializer);
+
+    // Of course you should never, ever block like this in real apps!!!
+    done.toCompletableFuture().get(5, TimeUnit.SECONDS);
```

Comment:
perhaps do it the right way instead?
`done.whenComplete...`

Reply:
Great point! Fixed.